automation_tools/__init__.py

```diff
@@ -1434,14 +1444,20 @@ def product_install(distribution, create_vm=False, certificate_url=None,
     if distribution == 'satellite6-cdn' and test_in_stage:
         execute(update_rhsm_stage, host=host)
 
-    # Subscribe for all distributions to install prerequisities and updates.
-    # Then let's unsubscribe just before calling ak_install.
-    execute(subscribe, host=host)
-
-    # Setting yum stdout log level to be less verbose
-    execute(set_yum_debug_level, host=host)
+    # Create custom OS beaker repo
+    if os.environ.get('OS_UPGRADE_REPO'):
+        os_upgrade_repo = os.environ.get('OS_UPGRADE_REPO')
+        execute(create_custom_repos, rhel_candidate=os_upgrade_repo, host=host)
 
-    execute(install_prerequisites, host=host)
+    # If NOT using an activationkey, subscribe to CDN and install some
+    # basic required packages.
+    if distribution != 'satellite6-activationkey':
```

Comment:
When we're not using an `activationkey`, then we must register to the CDN and get some prerequisite packages from it.